### PR TITLE
[StrategyDesignOptimizer] add StrategyDesignOptimizer

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot/octobot_backtesting_factory.py [hunk 1]
```diff
@@ -25,12 +25,14 @@ class OctoBotBacktestingFactory(octobot_class.OctoBot):
...
                  run_on_common_part_only=True,
-                 enable_join_timeout=True):
+                 enable_join_timeout=True,
+                 enable_logs=True):
         super().__init__(config)
         self.independent_backtesting = None
         self.log_report = log_report
         self.run_on_common_part_only = run_on_common_part_only
-        self.enable_join_timeout = enable_join_timeout
+        self.enable_join_timeout = enable_join_timeout,
```
Comment (Herklos) [inline octobot/octobot_backtesting_factory.py:34]:
```suggestion
        self.enable_join_timeout = enable_join_timeout
```

File: octobot/strategy_optimizer/strategy_design_optimizer.py [hunk 2]
```diff
@@ -0,0 +1,834 @@
...
+        start_queue = multiprocessing.Queue(len(run_data_by_hash))
+        done_queue = multiprocessing.Queue(len(run_data_by_hash))
+        for run_hash in run_data_by_hash:
+            start_queue.put(run_hash)
+        return {
+            self.START_QUEUE_KEY: start_queue,
+            self.DONE_QUEUE_KEY: done_queue
+        }
+
+    async def multi_processed_optimize(self,
```
Comment (Herklos) [inline octobot/strategy_optimizer/strategy_design_optimizer.py:132]: Can we split this method? It's hard to understand

File: octobot/strategy_optimizer/strategy_design_optimizer.py [hunk 3]
```diff
@@ -0,0 +1,834 @@
...
+            return left_operand > right_operand
+        if operator == "lower_or_equal_to":
+            return left_operand <= right_operand
+        if operator == "higher_or_equal_to":
+            return left_operand >= right_operand
+        if operator == "equal_to":
+            return left_operand == right_operand
+        if operator == "different_from":
+            return left_operand != right_operand
+        raise RuntimeError(f"Unknown operator: {operator}")
```
Comment (Herklos) [inline octobot/strategy_optimizer/strategy_design_optimizer.py:738]: I think we should move this to commons, we may need a similar feature somewhere else

File: octobot/strategy_optimizer/strategy_optimizer.py [hunk 4]
```diff
@@ -239,12 +245,15 @@ def print_report(self):
         self.logger.info(f"{best_result[CONFIG].get_result_string()} "
                          f"average trades count: {best_result[TRADES_IN_RESULT]:f}")
 
-    def get_overall_progress(self):
-        return int((self.run_id - 1) / self.total_nb_runs * 100) if self.total_nb_runs else 0
+    async def get_overall_progress(self):
```
Comment (Herklos) [inline octobot/strategy_optimizer/strategy_optimizer.py:248]: async ?

File: octobot/strategy_optimizer/strategy_optimizer.py [hunk 5]
```diff
@@ -239,12 +245,15 @@ def print_report(self):
...
+        return int((self.run_id - 1) / self.total_nb_runs * 100) if self.total_nb_runs else 0, 0
 
-    def is_in_progress(self):
+    async def is_in_progress(self):
```
Comment (Herklos) [inline octobot/strategy_optimizer/strategy_optimizer.py:251]: async ?

Comment (GuillaumeDSM) [reply octobot/strategy_optimizer/strategy_optimizer.py:251]: to keep the same interface as StrategyDesignOptimizer

**Pull request closed without merging**